libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -879,3 +882,72 @@ string YulUtilFunctions::suffixedVariableNameList(string const& _baseName, size_
 	}
 	return result;
 }
+
+string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const& _to)
+{
+	string functionName =
+		"convert_" +
+		_from.identifier() +
+		"_to_" +
+		_to.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		if (_from.category() == Type::Category::StringLiteral)
+		{
+			string const& data = dynamic_cast<StringLiteralType const&>(_from).value();
+			if (_to.category() == Type::Category::FixedBytes)
+			{
+				unsigned const numBytes = dynamic_cast<FixedBytesType const&>(_to).numBytes();
+				solAssert(data.size() <= 32, "");
+				Whiskers templ(R"(
+					function <functionName>() -> converted {
```

Comment:
Does it make sense to output a comment with the original string for easier debugging/output interpretation?

Reply:
Hm, we would need an escaping function at least...